packages/app/pages/index.tsx

```diff
@@ -292,22 +295,36 @@ const MembershipCard = (props: IMembershipCardProps) => {
             isOpen={shareIsOpen}
             setIsOpen={setShareIsOpen}
             title="Grant Access to Membership"
-            onConfirm={() => {
+            onConfirm={async () => {
               const contract = new web3.eth.Contract(
                 getAbiFromJson(passportFactoryJson)
               );
               contract.options.address =
                 contractAddressesByNetworkId[networkId]?.passportFactory || "";
-              return new Promise<void>((resolve, reject) =>
-                contract.methods
-                  .grantPassport(passport.address, userAddress)
-                  .send({ from: address })
-                  .on("receipt", () => {
-                    setUserAddress("");
-                    resolve();
-                  })
-                  .on("error", reject)
-              );
+              return (
+                userAddress.startsWith("0x")
+                  ? Promise.resolve(userAddress)
+                  : userAddress.endsWith(".eth")
+                  ? web3.eth.ens.getAddress(userAddress)
+                  : Promise.resolve("")
+              )
```

Comment:
Will abstract this method out when we merge your CSV PR and use it to convert each of those addresses

Reply:
great point, I swapped the order